app/features/syllabus/tools.py

```diff
@@ -0,0 +1,144 @@
+from typing import List, Tuple, Dict, Any
+#from io import BytesIO
+#from fastapi import UploadFile
+#from pypdf import PdfReader
+#from urllib.parse import urlparse
+#import requests
+import os
+import json
+import time
+
+from langchain_core.prompts import PromptTemplate
+#from langchain_core.runnables import RunnablePassthrough, RunnableParallel
+from langchain_core.output_parsers import JsonOutputParser
+from langchain_core.pydantic_v1 import BaseModel, Field, ValidationError
+from langchain_google_genai import GoogleGenerativeAI
+from langchain.chains.question_answering import load_qa_chain
+from langchain_core.exceptions import OutputParserException
+
+
+from services.logger import setup_logger
```

Comment:
Fetch these utility functions from the `app` folder.

Reply:
fixed it